src/client/opamCommands.ml

```diff
@@ -330,23 +330,19 @@ let init =
       | None ->
         OpamFilename.of_string (OpamStd.Sys.guess_dot_profile shell)
     in
+    let init_config =
+      get_init_config ~no_sandboxing
+        ~no_default_config_file:no_config_file ~add_config_file:config_file
+    in
     if already_init then
       if reinit then
-        let init_config =
-          get_init_config ~no_sandboxing
-            ~no_default_config_file:no_config_file ~add_config_file:config_file
-        in
         OpamClient.reinit ~init_config ~interactive ~dot_profile
           ?update_config ?env_hook ?completion
           (OpamFile.Config.safe_read config_f) shell
       else
         OpamEnv.setup root
```

Comment:
This wasn't really duplicate: we don't need to call it in this case, where it is not needed (because it has some side-effects like printed messages and possibly errors)

Reply:
ok!